src/dns/aardvark.rs

```diff
@@ -0,0 +1,256 @@
+                                                } else {
+                                                    network_gateway_v6 = gateway.to_string();
+                                                }
+                                            }
+
+                                            if container_ip.is_ipv4() {
+                                                container_ip_v4 = container_ip.to_string();
+                                            } else {
+                                                container_ip_v6 = container_ip.to_string();
+                                            }
+
+                                            result.push(AardvarkEntry {
+                                                network_name: network_name.clone(),
+                                                network_gateway_v4,
+                                                network_gateway_v6,
+                                                container_id: container_id.clone(),
+                                                container_ip_v6,
+                                                container_ip_v4,
+                                                container_name: Vec::from([container_name.clone()]),
+                                            });
+                                        }
+                                    }
+                                    None => continue,
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        result
+    }
+
+    pub fn delete_entry(self, container_id: String, network_name: String) -> Result<()> {
+        let path = Path::new(&self.config).join(network_name);
+        let mut file = File::open(&path)?;
+        let reader = BufReader::new(&mut file);
+
+        let lines: Vec<_> = reader
+            .lines()
+            .map(|l| l.expect("Couldn't read a line"))
+            .collect();
+
+        let mut file = File::create(&path)?;
+
+        for line in &lines {
+            if line.contains(&container_id) {
+                if lines.len() <= 3 {
+                    // delete the file and return
+                    // since there is nothing in the network
+                    fs::remove_file(&path)?;
+                    return Ok(());
+                }
+                continue;
+            }
+            file.write_all(line.as_bytes())?;
+            file.write_all(b"\n")?;
+        }
+        Ok(())
+    }
+
+    pub fn delete_from_netavark_entries(
+        self,
+        network_options: types::NetworkOptions,
+    ) -> Result<()> {
+        let mut modified = false;
+        let container_id = network_options.container_id;
+        for (key, network) in network_options.network_info {
+            if network.dns_enabled {
+                modified = true;
+                self.clone().delete_entry(container_id.clone(), key)?;
```

Comment:
Why do we have to clone self?

Reply:
`self` is used again later in this iteration and `self` does not implements `copy` trait.

Reply:
fixed with `&mut`